### ISSUE TITLE
Finetuning BART on CNN/DM : Why not truncating long articles ?

### ISSUE THREAD

**Issue by Colanim** [pytorch/fairseq#1414]
In the [README for finetuning BART on CNN/DM](https://github.com/pytorch/fairseq/blob/master/examples/bart/README.cnn.md), the command for finetuning BART does not specify  
`--truncate-source`.

By running the README command, I have following warning :

> WARNING: 86040 samples have invalid sizes and will be skipped [...]

But if I specify the option `--truncate-source`, the warning become :

> WARNING: 5 samples have invalid sizes and will be skipped [...]

---

**Why the option `--truncate-source` is not used ?**  
It feels like skipping 80k samples is detrimental for performance, and other existing architectures always truncate article if it's too long..

@ngoyal2707 @yinhanliu

**Comment by yinhanliu** [comment]
it is truncated in the code, 

https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/fairseq/tasks/translation.py#L59

They should result the same.

**Comment by astariul** [comment]
Thanks for the answer @yinhanliu 

The code you link appear to be in the scope of this `if` :
https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/fairseq/tasks/translation.py#L55-L62

So if the option `--truncate-source` is not specified, we never enter this `if`, right ?

**Comment by yinhanliu** [comment]
@astariul thanks for pointing this out. Apparently Naman changed code when he pushed code than what I have when I fine-tuned cnn/dm. But Naman is on vacation now. I can only tell what I did to get the number in the paper. I truncated all the source into 1024 -4  tokens.  then make it into <BOS> <truncated_source> < EOS> <MASK> < EOS>. When we tried to release the code, Naman tried to remove <MASK> and <EOS>, and it turned out to not make a difference. Either way, you have to truncate the source to fit it into 1024. I probably will have time later next week to take a look of this code to see where the truncation happened without --truncate-source.
Thanks for pointing this out to me.

**Comment by huihuifan** [comment]
closing after Yinhan's answer

**Comment by Colanim** [comment]
I don't think it should be closed yet : we are still waiting for a confirmation from @ngoyal2707 .

**Comment by ngoyal2707** [comment]
@Colanim Yes, we truncated the long sources, in fact, that's the very reason I introduced the `--truncate-source` arg in the code. 
Just a missing arg in readme, will add the fix. Thanks for pointing out @Colanim
